apps/communications/dialer/js/call_log.js

```diff
@@ -10,7 +10,7 @@ var CallLog = {
   _headersInterval: null,
   _empty: true,
   _dbupgrading: false,
-  _contactCache: true,
+  _contactCache: false,
```

Comment:
Good point, this would be racy if the cache was valid and we would call ``_removeContact()`` just before ``_contactCache`` was set to true. I'll try to modify the patch to wait in this particular scenario.